core/src/main/java/hudson/model/Job.java

```diff
@@ -316,8 +318,56 @@ public boolean isBuilding() {
      */
     public boolean isLogUpdated() {
         RunT b = getLastBuild();
-        return b!=null && b.isLogUpdated();
-    }    
+        return b != null && b.isLogUpdated() || isCurrentWorkUnit();
+    }
+
+    /**
+     * Returns {@code true} if the job is a current work unit on any node.
+     *
+     * @return {@code true} if the job is a current work unit on any node.
+     * @since FIXME
+     */
+    public boolean isCurrentWorkUnit() {
+        if (this instanceof SubTask) {
+            // if you are not a SubTask then you cannot be a work unit on any node.
+            final Jenkins jenkins = Jenkins.getInstance();
+            if (jenkins != null) {
+                if (isCurrentWorkUnit(jenkins)) {
+                    return true;
+                }
+                for (Node n : jenkins.getNodes()) {
+                    if (isCurrentWorkUnit(n)) {
+                        return true;
+                    }
+                }
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Returns {@code true} if the job is a current work unit on the specified node.
+     *
+     * @param node the specified node
+     * @return {@code true} if the job is a current work unit on the specified node.
+     */
+    private boolean isCurrentWorkUnit(Node node) {
+        final Computer c = node.toComputer();
+        if (c == null) return false;
```

Comment:
Not when in Queue.maintain() which is the only time it matters

Sent from my iPhone

> On 21 Sep 2015, at 8:38 a.m., Oleg Nenashev <notifications@github.com> wrote:
> 
> In core/src/main/java/hudson/model/Job.java:
> 
> > +                    }
> > +                }
> > +            }
> > +        }
> > +        return false;
> > +    }
> > +
> > +    /**
> > +     * Returns {@code true} if the job is a current work unit on the specified node.
> > +     *
> > +     * @param node the specified node
> > +     * @return {@code true} if the job is a current work unit on the specified node.
> > +     */
> > +    private boolean isCurrentWorkUnit(Node node) {
> > +        final Computer c = node.toComputer();
> > +        if (c == null) return false;
> There is a risk of false-negative result if nodes are being reconfigured in parallel with checks.
> 
> —
> Reply to this email directly or view it on GitHub.
> 
> -- 
> You received this message because you are subscribed to the Google Groups "engineering-code-reviews" group.
> To unsubscribe from this group and stop receiving emails from it, send an email to engineering-code-reviews+unsubscribe@cloudbees.com.
> To post to this group, send email to engineering-code-reviews@cloudbees.com.
> To view this discussion on the web visit https://groups.google.com/a/cloudbees.com/d/msgid/engineering-code-reviews/jenkinsci/jenkins/pull/1834/r39946328%40github.com.